main.tf

```diff
@@ -260,3 +260,23 @@ resource "aws_security_group_rule" "cidr_ingress" {
   cidr_blocks       = var.allowed_cidr_blocks
   security_group_id = local.rds_security_group_id
 }
+
+################################################################################
+# Cluster Endpoints
+################################################################################
+
+resource "aws_rds_cluster_endpoint" "custom" {
+  for_each = var.create_cluster && var.create_cluster_custom_endpoints ? var.cluster_custom_endpoints : {}
+
+  cluster_identifier          = element(concat(aws_rds_cluster.this.*.id, [""]), 0)
+  cluster_endpoint_identifier = lower(each.key)
+  custom_endpoint_type        = each.value
+
+  lifecycle {
+    ignore_changes = [excluded_members, static_members]
```

Comment:
I don't yet know what the ramifications of this are. we don't have these attributes set here, and yet we are ignoring them from future changes. we generally try to avoid the use of the `ignore_changes` lifecycle hook because it cannot be parameterized. I will have to do some tests to see what makes the most sense for this

Reply:
I'll do a couple of tests on my own.  I am currently using this version and will check the behavior without ignoring

Reply:
We can use `static_members` for case then we need include **only** instances created by Terraform (by this module)
If we need reader endpoint only with nodes managed from iac. 

And we can use `excluded_members` for case then we need exclude **all** instances created by this module. 
For some custom/temporary endpoints

By default both will be empty 

```hcl
var.cluster_custom_endpoints_only_static_members = false
var.cluster_custom_endpoints_all_excluded_members = false
```